pkg/task/task.go

```diff
@@ -3,29 +3,29 @@ package task
 import "github.com/google/uuid"
 import "time"
 
-// TaskState (kind: int) represents the last known state of a task.
+// TaskState (kind: string) represents the last known state of a task.
 // A task can be in one of three states
 // StateScheduled: this is the initial state of the task when it enters the queue.
 // StateProcessing: once work begins on the task, it is put into this state.
 // StateComplete: work is no longer being done on this task. client should check task result.
-type TaskState int
+type TaskState string
```

Comment:
Nitpick: this is already in the `task` package, maybe just rename to `State`? No need to `task.TaskState`.